pybamm/expression_tree/symbol.py

```diff
@@ -894,46 +892,40 @@ def new_copy(self):
         obj._print_name = self.print_name
         return obj
 
-    @property
+    @cached_property
     def size(self):
         """
         Size of an object, found by evaluating it with appropriate t and y
         """
-        try:
-            return self._saved_size
-        except AttributeError:
-            self._saved_size = np.prod(self.shape)
-            return self._saved_size
+        self._saved_size = np.prod(self.shape)
+        return self._saved_size
```

Comment:
should just be `return np.prod(self.shape)`, see https://github.com/pybamm-team/PyBaMM/pull/2465#issuecomment-1319085195